plutus-core/plutus-ir/src/PlutusIR/TypeCheck/Internal.hs

```diff
@@ -375,39 +386,52 @@ withVarsOfBinding r (DatatypeBind _ dt) k = do
     foldr normRenameScope k structorDecls
     where
       -- normalize, then introduce the vardecl to scope
-      normRenameScope :: VarDecl TyName Name uni fun (Provenance a)
-                      -> TypeCheckM uni fun c e res -> TypeCheckM uni fun c e res
+      normRenameScope :: VarDecl TyName Name uni fun (Provenance ann)
+                      -> TypeCheckT uni fun cfg m a -> TypeCheckT uni fun cfg m a
       normRenameScope v acc = do
           normRenamedTy <- normalizeTypeM $ _varDeclType v
           withVar (_varDeclName v) (void <$> normRenamedTy) acc
 
 
-withVarsOfBindings :: (Foldable t, HasUniApply uni) => Recursivity -> t (Binding TyName Name uni fun a)
-                   -> TypeCheckM uni fun c e res -> TypeCheckM uni fun c e res
+withVarsOfBindings
+    :: (MonadNormalizeType uni m, Foldable t)
+    => Recursivity
+    -> t (Binding TyName Name uni fun ann)
+    -> TypeCheckT uni fun cfg m a
+    -> TypeCheckT uni fun cfg m a
 withVarsOfBindings r bs k = foldr (withVarsOfBinding r) k bs
 
 -- | Scope a typechecking computation with the given binding's newly-introducing type (if there is one)
-withTyVarsOfBinding :: Binding TyName name uni fun ann -> TypeCheckM uni fun c e res -> TypeCheckM uni fun c e res
+withTyVarsOfBinding
+    :: Monad m
+    => Binding TyName name uni fun ann
+    -> TypeCheckT uni fun cfg m a
+    -> TypeCheckT uni fun cfg m a
```

Comment:
It was sometimes `res`, sometimes `a` in this file. Made it `a` everywhere. I believe fully polymorphic things with no other meaning than "it can be absolutely anything" it makes sense to use a single letter, while, say, annotations do have semantics attached to them.